spec/classes/certs_qpid_router_spec.rb

```diff
@@ -1,11 +1,13 @@
 require 'spec_helper'
 
 describe 'certs::qpid_router' do
-  let :facts do
-    on_supported_os['redhat-7-x86_64']
-  end
+  on_supported_os.each do |os, facts|
+    context "on #{os}", if: facts[:operatingsystemmajrelease] == '7' do
+      let :facts do facts end
```

Comment:
Slightly easier to read:
```suggestion
      let(:facts) { facts }
```